### ISSUE TITLE
[bug] `<rh-tabs>`: panel `:focus` and `:focus-within` causing multiple concurrent outlines

### ISSUE THREAD

**Issue by hellogreg** [RedHat-UX/red-hat-design-system#2970]
Our tab panels currently receive focus outlines when:

1. The panel itself has keyboard focus (good)
2. When clicked within via the mouse ([not necessary](https://www.w3.org/WAI/WCAG22/Understanding/focus-visible.html), but maybe we're doing it deliberately?)
3. When any other element within has keyboard focus (bad).

 
I think issue 3 above should be fixed, so only one item on the page has a visible focus state at a time. And we can discuss 2.

Here's a screencap of dual focus. Note the outline on both the panel and the link within.

<img width="816" height="325" alt="Tab panel with two focus outlines" src="https://github.com/user-attachments/assets/eabae1e0-77bf-48d6-ad24-84c689c92ef7" />

[Example page here](https://www.redhat.com/en/about/brand/standards/color#how-we-use-color)

It can get even more complicated if someone is using tabs within tabs, which I guess has been known to happen!

Looks like the relevant code is in [rh-tab-panel.css](https://github.com/RedHat-UX/red-hat-design-system/blob/9d6ee86c59aa1052e3d767d7015d030b2d095987/elements/rh-tabs/rh-tab-panel.css#L12).

Potential solution: replace `:is(:focus, :focus-within)` with `:focus-visible`, unless there's a reason we prefer `:focus` to `:focus-visible`.
